Refs #34140 -- Applied rst code-block to non-Python examples.

carltongibson opened the pull request:
Looking to push forward the blacken-docs effort, applied the required `code-blocks` throughout. 

* Possibly I missed some. 
* Possibly we can tweak languages later. 

But I really don't want this to go stale. 😅 — so can we decide quickly please. 

Note: we loose the ability to set default for a page say — this was used in template docs, and in the ORM docs, where there are lots of `pycon` examples back to back often. 

I feel a bit sad about this, but I think it's probably worth doing to get the blacken-docs into play. (Eye on the goal 😜)
We'd then just have `::` for Python throughout the project. 

Moving is a one-time cost, that I've paid here. 

Also, use of the override wasn't consistent. I think we're likely in a better world the other side of this change if we go for it. 

@adamchainz Can I ask you to review and confirm we're going for this. (On balance I'm probably for it but I'd like you to champion it assuming you're +1, which I think you are.) 

//cc @jvzammit @pauloxnet — hopefully this frees #16496 to be easier.

jvzammit commented on docs/faq/troubleshooting.txt:
```diff
@@ -36,7 +36,9 @@ I'm getting a ``UnicodeDecodeError``. What am I doing wrong?
-output generally looks like this::
+output generally looks like this:
+
+.. code-block:: pytb
```
Nice, I didn't even know about `pytb`!

jvzammit commented on docs/howto/windows.txt:
```diff
@@ -28,7 +26,9 @@ Download the executable installer and run it. Check the boxes next to "Install
-matches the version you installed by executing::
+matches the version you installed by executing:
+
+.. code-block:: doscon
```
TIL `doscon` as well 👏

jvzammit commented on docs/ref/contrib/gis/geoip2.txt:
```diff
@@ -33,6 +33,8 @@ Example
 
 Here is an example of its usage::
```
```suggestion
Here is an example of its usage:
```

jvzammit commented on docs/howto/csrf.txt:
```diff
@@ -227,6 +227,8 @@ If, for some reason, you *want* the test client to perform CSRF
 checks, you can create an instance of the test client that enforces
 CSRF checks::
```
```suggestion
CSRF checks:
```

jvzammit commented on docs/ref/contrib/postgres/expressions.txt:
```diff
@@ -24,6 +24,8 @@ an SQL ``GROUP BY`` clause to build the list of values.
 For example, if you want to annotate all related books to an author as JSON
 objects::
```
```suggestion
objects:
```

jvzammit replied:
@carltongibson Merge this one as well?

carltongibson replied:
Just didn't get to it yet 😅

jvzammit reviewed (changes requested):
Thanks for the huge amount of grunt-but-not-so-grunt work in this @carltongibson !!

This indeed will make the commits in PRs to get to "blacken-docs" cleaner. I think you had suggested this but I only understand what you meant now 🤦‍♂️ 

In any case thanks! I spotted some very, very minor double-colon "typos". And if there's anything I didn't spot I think should be easy-to-fix later on 👍 

Thanks again!

pauloxnet commented on docs/intro/reusable-apps.txt:
```diff
@@ -256,9 +258,11 @@ this. For a small app like polls, this process isn't too difficult.
+   .. code-block:: text
 
-    recursive-include docs *
+     recursive-include docs *
```
```suggestion
      recursive-include docs *
```

pauloxnet commented on docs/ref/contrib/postgres/aggregates.txt:
```diff
@@ -349,16 +359,19 @@ field or an expression returning a numeric data. Both are required.
-We will use this example table::
+We will use this example table:
+
+.. code-block:: text
```
```suggestion
.. code-block:: markdown
```
Could it be markdown ?

carltongibson replied:
Meh. Argh! — I'd like to punt on that to a separate cleanup please. (IDK 🤯)

pauloxnet replied:
It was only a question. I agree with you.

pauloxnet commented on docs/ref/django-admin.txt:
```diff
@@ -91,7 +93,9 @@ Uses the :doc:`system check framework </ref/checks>` to inspect the entire
-providing a list of app labels as arguments::
+providing a list of app labels as arguments:
+
+.. console::
```
if it is equivalent to `code-block::shell` perhaps it would be better to use the same mode everywhere.

carltongibson replied:
It's not. It gives a special windows/*nixy tabbed interface. Maybe there's a case for making some extra cases use it, but I'd like to address that separately, as it needs thinking about. (e.g. The GIS docs are very OS specific, and so on. )

pauloxnet replied:
I didn't know about this special behavior of `console`.
It's perfectly fine to leave it at that.

pauloxnet commented:
it's an amazing job. I've only added a few comments, most of which are questions rather than corrections.

carltongibson commented on docs/howto/custom-management-commands.txt:
```diff
@@ -12,7 +12,9 @@ command for the ``polls`` application from the
-whose name doesn't begin with an underscore. For example::
+whose name doesn't begin with an underscore. For example:
+
+.. code-block:: text
```
I don't know what `tree` gives on your system but I think it's by-hand 🤔


```
% tree -L 1
.
├── Makefile
├── README.rst
├── _build
├── _ext
├── _theme
├── conf.py
├── contents.txt
├── faq
├── glossary.txt
├── howto
├── index.txt
├── internals
├── intro
├── make.bat
├── man
├── misc
├── ref
├── releases
├── requirements.txt
├── spelling_wordlist
└── topics

12 directories, 9 files
```

pauloxnet replied:
Right! I'm closing this.

pauloxnet reviewed (approved):
@carltongibson thanks for all your answers.
For me this PR can be merged.

felixxm reviewed (commented):
@carltongibson Thanks for you humongous work :1st_place_medal: :mountain: 

I don't like bulk updates, but, as far as I'm aware, we have no choice 😐 :

Should we also remove (currently unused :thinking: ) `highlight` directives?:

- `howto/deployment/wsgi/uwsgi.txt`: `.. highlight:: bash`
- `howto/deployment/wsgi/gunicorn.txt`: `.. highlight:: bash`
- `howto/deployment/asgi/daphne.txt`: `.. highlight:: bash`
- `howto/deployment/asgi/hypercorn.txt`: `.. highlight:: bash`
- `howto/deployment/asgi/uvicorn.txt`: `.. highlight:: bash`
- `internals/contributing/writing-code/unit-tests.txt`: `.. highlight:: python`
- `internals/contributing/triaging-tickets.txt`: `.. highlight:: console`
- `internals/howto-release-django.txt`: `.. highlight:: console`
- `ref/contrib/gis/install/index.txt`: `.. highlight:: console`
- `releases/1.0-porting-guide.txt`: `.. highlight:: python`
- `topics/security.txt`: `.. highlight:: html+django`
- `topics/i18n/translation.txt`: `.. highlight:: python`
- `topics/i18n/translation.txt`: `.. highlight:: javascript`
- `topics/i18n/translation.txt`: `.. highlight:: python`
- `topics/i18n/translation.txt`: `.. highlight:: python`
- `topics/templates.txt`: `.. highlight:: python`
- `topics/forms/modelforms.txt`: `.. highlight:: python`
- `topics/cache.txt`: `.. highlight:: python`

carltongibson commented:
Thanks @felixxm! 

> I don't like bulk updates, but, as far as I'm aware, we have no choice 😐 :

Yep, me neither. They're recipe for trouble... — but yes, necessary if we're going to proceed here. 

I applied all the suggestions and removed the `highlight` directives, and rebased.

carltongibson commented:
Thanks @pauloxnet 😊 it's not that I think it's not worth it, but I'm pulled both ways (ambivalent) 😛

smithdc1 commented:
Presumably this and future related commits would be good candidates to add to the git-blame-ignore-revs file? 🤔

felixxm commented on docs/intro/tutorial05.txt:
```diff
@@ -267,7 +269,9 @@ past:
-and run the test again::
+and run the test again:
+
+.. code-block:: pytb
```
It it not changed :thinking:

carltongibson replied:
Hmm. I clicked the commit button 🤔

Computers! 🤪

(If you're looking to merge, please just edit and do! Otherwise we can resolve later. 💝)

felixxm commented:
> Presumably this and future related commits would be good candidates to add to the git-blame-ignore-revs file?

Yes, I'll add it to the `.git-blame-ignore-revs`.

adamchainz commented:
A little late to the party here, but all LGTM 👍
